specs/ExecuteScript.md

```diff
@@ -0,0 +1,222 @@
+                                            [](HRESULT error, PCWSTR result) -> HRESULT {
+                                                return S_OK;
+                                            })
+                                            .Get());
+                                    return S_OK;
+                                })
+                                .Get(),
+                            NULL);
+                    }
+                    return S_OK;
+                })
+                .Get(),
+            NULL));
+    }
+```
+
+# API Notes
+See [API Details](#api-details) section below for API reference.
+
+# API Details
+## IDL
+```c#
+/// WebView2Frame provides direct access to the iframes information and use.
+[uuid(423ba05f-55dd-4c50-8339-22afd009ed31), object, pointer_default(unique)]
+interface ICoreWebView2ExperimentalFrame : IUnknown {
+  /// Add an event handler for the `NavigationStarting` event.
+  /// A frame navigation will raise a `NavigationStarting` event and
+  /// a `CoreWebView2.FrameNavigationStarting` event. All of the
+  /// `FrameNavigationStarting` event handlers will be run before the
+  /// `NavigationStarting` event handlers. All of the event handlers share a
+  /// common `NavigationStartingEventArgs` object. Whichever event handler is
+  /// last to change the `NavigationStartingEventArgs.Cancel` property will
+  /// decide if the frame navigation will be cancelled. Redirects raise this
+  /// event as well, and the navigation id is the same as the original one.
+  ///
+  /// You may block corresponding navigations until the event handler returns.
+  ///
+  /// \snippet ScenarioWebViewEventMonitor.cpp NavigationStarting
+  HRESULT add_NavigationStarting(
+      [in] ICoreWebView2FrameNavigationStartingEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+
+  /// Remove an event handler previously added with `add_NavigationStarting`.
+  HRESULT remove_NavigationStarting(
+      [in] EventRegistrationToken token);
+
+  /// Add an event handler for the `ContentLoading` event.  `ContentLoading`
+  /// triggers before any content is loaded, including scripts added with
+  /// `AddScriptToExecuteOnDocumentCreated`.  `ContentLoading` does not trigger
+  /// if a same page navigation occurs (such as through `fragment`
+  /// navigations or `history.pushState` navigations).  This operation
+  /// follows the `NavigationStarting` and precedes `NavigationCompleted` events.
+  HRESULT add_ContentLoading(
+      [in] ICoreWebView2FrameContentLoadingEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+
+  /// Remove an event handler previously added with `add_ContentLoading`.
+  HRESULT remove_ContentLoading(
+      [in] EventRegistrationToken token);
+
+  /// Add an event handler for the `NavigationCompleted` event.
+  /// `NavigationCompleted` runs when the CoreWebView2Frame has completely
+  /// loaded (`body.onload` runs) or loading stopped with error.
```

Comment:
I'm guessing we raise NavigationCompleted immediatley *before* `body.unload` runs? Or does it get raised concurrently with `body.unload`? (I.e., page's unload handlers run in parallel while the NavigationCompleted handler is running.)

```suggestion
  /// loaded (immediately before `body.onload` runs) or loading stopped with error.
```

Reply:
Update docs to say if body.onload runs immediately before or immediately after or concurrent?